lib/cocina/models/folio_catalog_link.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Cocina
+  module Models
+    # A linkage between an object and a Folio catalog record
+    class FolioCatalogLink < Struct
+      # Catalog that is the source of the linked record.
+      # example: folio
+      attribute :catalog, Types::Strict::String.enum('folio', 'previous folio')
+      # Only one of the catkeys should be designated for refreshing. This means that this key is the one used to pull metadata from the catalog if there is more than one key present.
+      attribute :refresh, Types::Strict::Bool.default(false)
+      # Record identifier that is unique within the context of the linked record's catalog.
+      attribute :catalogRecordId, MigratedFromSymphonyIdentifier | MigratedFromVoyagerIdentifier | CreatedInFolioIdentifier
```

Comment:
Building on the past couple PRs, this PR can now use a union type of generated types.